noir/noir-repo/docs/docs/noir/concepts/functions.md

```diff
@@ -184,14 +184,16 @@ See [Lambdas](./lambdas.md) for more details.
 
 Attributes are metadata that can be applied to a function, using the following syntax: `#[attribute(value)]`.
 
-Supported attributes include:
+A few supported attributes include:
 
 - **builtin**: the function is implemented by the compiler, for efficiency purposes.
 - **deprecated**: mark the function as _deprecated_. Calling the function will generate a warning: `warning: use of deprecated function`
 - **field**: Used to enable conditional compilation of code depending on the field size. See below for more details
 - **oracle**: mark the function as _oracle_; meaning it is an external unconstrained function, implemented in noir_js. See [Unconstrained](./unconstrained.md) and [NoirJS](../../reference/NoirJS/noir_js/index.md) for more details.
 - **test**: mark the function as unit tests. See [Tests](../../tooling/testing.md) for more details
 
+See the Noir compiler for the full list of supported attributes [here](https://github.com/noir-lang/noir/blob/master/compiler/noirc_frontend/src/lexer/token.rs#L631) (the `attribute` match inside the `Attribute` implementation).
```

Comment:
As this points to a line number, this could potentially change what it is pointing to, it may be better to make this a permalink or maybe use the #include_code macros (although im not sure that has been used in the noir compiler before)?

Reply:
The description that follows in brackets was intended to help them find the Attribute, even if the line number loses precision over time. I don't think locking the commit is ideal since the attributes may change, so removing the line number then leaves the dev to find definitions in the 1000 line file.
I'll add a more specific line to search for eg "`let attribute = match &word_segments[..] {` at the time of writing"